src/validation/rules.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Debug;
+
+use crate::validation::{
+    data::*,
+    violation::{self, *},
+};
+
+mod file_name;
+mod ident_casing;
+mod package_name;
+
+pub use self::{file_name::*, ident_casing::*, package_name::*};
+
+/// Collection of rules.
+pub type RuleSet = Vec<Box<dyn Rule>>;
+
+/// Rule to enforce for buffrs packages.
+pub trait Rule: Debug {
+    /// Name of this rule.
+    ///
+    /// Defaults to the name of the type of this rule.
+    fn rule_name(&self) -> &'static str {
```

Comment:
can we require all of these to be const?

Reply:
I would actually, make most of them associated constants, rather than functions.

Reply:
it should, pretty sure

Reply:
can we please try out?

Reply:
Do we need dyn Rule? 

As said, i would make them statically typed make enums to list them explicitly

Reply:
wont do for now